pkgs/development/libraries/cpp-utilities/default.nix

```
@@ -0,0 +1,27 @@
+{ stdenv, fetchFromGitHub, cmake, cppunit }:
+
+stdenv.mkDerivation rec {
+  pname = "cpp-utilities";
+  version = "4.17.1";
+
+  src = fetchFromGitHub {
+    owner = "Martchus";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "12088cwg3jbqipmbn4843w1cgxi1q6vwx47gy042rkfvbk6azhxl";
+  };
+
+  nativeBuildInputs = [ cmake ];
+  checkInputs = [ cppunit ];
+  # Otherwise, tests fail since the resulting shared object libc++utilities.so is only available in PWD of the make files
+  checkFlagsArray = [ "LD_LIBRARY_PATH=$(PWD)" ];
+  doCheck = true;
+
+  meta = with stdenv.lib; {
+    homepage = "https://github.com/Martchus/cpp-utilities";
+    description = "Common C++ classes and routines used by @Martchus' applications featuring argument parser, IO and conversion utilities";
+    license = licenses.gpl2;
+    maintainers = with maintainers; [ doronbehar ];
+    platforms   = platforms.all;
```

the repository has a unicode filename [`testfiles/täst.txt`](https://github.com/Martchus/cpp-utilities/blob/81a3d4fa9477ea7681b9c8e6ddd87b5f5dc996c9/testfiles/t%C3%A4st.txt), which causes the source the src have a different hash on darwin HFS+ filesystem, because unicode normalization.

If you are not interested on darwin, you can only specify linux for now:
```suggestion
    platforms = platforms.linux;
```
but if you are interested in darwin you can add [`extraPostFetch` to fetchFromGitHub](https://github.com/nixos/nixpkgs/blob/def7d562e01430ab48f4ab84a6363bfdbe6e584a/pkgs/applications/version-management/gitea/default.nix#L20) to remove the offending file.

Besides that, LGTM. Also, please squash the commits

---

Actually, the author explicitly states that the library does support being compiled on OSx so it would have been nice to fix this small issue somehow. It should be possible to add an `if` statement and change the hash based on `stdenv.isDarwin` right?

I'll commit this suggestion and hopefully someone using Darwin will find a workaround in the future. I don't have the means to test workarounds for this anyway.